### ISSUE TITLE
[Java] Memory issue about FuryInputStream

### ISSUE THREAD

**Issue by yihwang6** [apache/incubator-fury#1576]
In #1551, I noticed that a growBuffer method was added to FuryInputStream, which continuously increases the size of the buffer. This operation may potentially give rise to memory-related concerns due to the large buffer size. It might be better to consider implementing the buffered streaming reading with a fixed-size buffer.

**Comment by chaokunyang** [comment]
`org.apache.fury.io.FuryInputStream#shrinkBuffer` will be invoked after the deserialziation finished

**Comment by yihwang6** [comment]
Ok, but there may still be a large buffer size when deserializing a large object.

**Comment by chaokunyang** [comment]
Yes, that is possible. We don't release buffer before we finished the deserialization. One case I can imagine is that when buffer close to int max size, we can't multiple 1.5.

**Comment by chaokunyang** [comment]
Would you like to help optimize this?
